Assignment/Assignment.Tests/SampleDataTests.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+
+namespace Assignment.Tests
+{
+	[TestClass]
+	public class SampleDataTests
+	{
+		[TestMethod]
+		public void CSVRows_GetsAllTheRows_TheRowsMatch()
+		{
+			SampleData testData = new ();
+
+			Assert.AreEqual("1,Priscilla,Jenyns,pjenyns0@state.gov,7884 Corry Way,Helena,MT,70577", testData.CsvRows.First());
+			Assert.AreEqual("15,Phillida,Chastagnier,pchastagniere@reference.com,1 Rutledge Point,Spokane,WA,99021", testData.CsvRows.ElementAt(14));
+			Assert.AreEqual("25,Jedd,Boissier,jboissiero@cbsnews.com,1 Arrowood Crossing,San Diego,CA,96101", testData.CsvRows.ElementAt(24));
+			Assert.AreEqual("50,Claudell,Leathe,cleathe1d@columbia.edu,30262 Steensland Way,Newport News,VA,87930", testData.CsvRows.Last());
+			Assert.AreEqual(50, Enumerable.Count(testData.CsvRows));
+		}
+
+		[TestMethod]
+		public void SortsRowsByState_CountsNumberOfUniqueStates_SuccessWhen27States()
+        {
+			SampleData testData = new();
+			Assert.AreEqual(27,Enumerable.Count(testData.GetUniqueSortedListOfStatesGivenCsvRows()));
+        }
+
+		[TestMethod]
+		public void SortsRowsByState_ChecksIfStatesAreUnique_SuccessDistinctIsTrue()
+        {
+			SampleData testData = new();
+			Assert.IsTrue(testData.CsvRows.SequenceEqual(testData.CsvRows.Distinct()));
+        }
+
+		[TestMethod]
+		public void SortRowsByState_ChecksIfStatesInOrder_SuccessWhenInOrder()
+        {
+			SampleData testData = new();
+			Assert.AreEqual(testData.GetUniqueSortedListOfStatesGivenCsvRows().First(), (testData.GetUniqueSortedListOfStatesGivenCsvRows().OrderBy(item=>item).First()));
+			Assert.AreEqual(testData.GetUniqueSortedListOfStatesGivenCsvRows().Last(), (testData.GetUniqueSortedListOfStatesGivenCsvRows().OrderBy(item => item).Last()));
+			Assert.IsTrue(testData.GetUniqueSortedListOfStatesGivenCsvRows().SequenceEqual(testData.GetUniqueSortedListOfStatesGivenCsvRows().OrderBy(item => item)));
+
+		}
+
+        [TestMethod]
+        public void People_ContainsAllFields()
+        {
+			SampleData testData = new();
+			var person = testData.People.First();
+			string test = 
+				person.FirstName + "," + 
+				person.LastName + "," + 
+				person.EmailAddress + "," + 
+				person.Address.StreetAddress + "," + 
+				person.Address.City + "," + 
+				person.Address.State + "," + 
+				person.Address.Zip;
+			Assert.AreEqual("Arthur,Myles,amyles1c@miibeian.gov.cn,4718 Thackeray Pass,Mobile,AL,37308", test);
+        }
+
+        [TestMethod]
+        public void People_GetsAllRows_MatchSortedFirstLast()
+        {
+			SampleData testData = new();
+			Assert.AreEqual("Arthur,Myles", $"{testData.People.First().FirstName},{testData.People.First().LastName}");
+			Assert.AreEqual("Amelia,Toal", $"{testData.People.Last().FirstName},{testData.People.Last().LastName}");
+			Assert.AreEqual(50, Enumerable.Count(testData.People));
```

Comment:
I think this is fine since _Enumerable.Count_ is technically just a static method, since it's an extension method, you're really just calling _testData.People.Count_ .

```suggestion
			Assert.AreEqual(50, testData.People.Count()));
```